code/modules/reagents/chemistry/reagents/cat2_medicine_reagents.dm

```diff
@@ -132,43 +132,43 @@
 /******TOXIN******/
 /*Suffix: -iver*/
 
-#define FIZ_WEAKRATE	0.005
-
 /datum/reagent/medicine/C2/fiziver //fiz = phys ok?
   name = "Fiziver"
-  description = "An antitoxin that temporarily weakens the user, making them susceptible to other forms of damage. Weakness and toxin healing scales with exposure."
-  overdose_threshold = 11 // Scales every damage so naturally quite abusable. Luckily the OD purges the chem which resets the damage mod *wink*
-  var/reset_mods = 0 //amount, not bool
+  description = "An antitoxin that temporarily weakens the user, making them susceptible to other forms of damage. Weakness and toxin healing scales with length of exposure."
+  overdose_threshold = 11
+  metabolization_rate = 0.25 * REAGENTS_METABOLISM //so that the weakness from a 10u pill will last for around 3 minutes or so
+  var/weak_mod = 1
 
 /datum/reagent/medicine/C2/fiziver/on_mob_life(mob/living/carbon/human/M)
-	var/maths = FIZ_WEAKRATE * current_cycle
-	reset_mods += maths
 	var/datum/physiology/phis = M.physiology
-	phis.brute_mod += maths
-	phis.burn_mod += maths
-	phis.oxy_mod += maths
-	phis.stamina_mod += maths
-	M.adjustToxLoss(round(reset_mods*-1000,0.01)) //Math is fun!
+	phis.brute_mod /= weak_mod
+	phis.burn_mod /= weak_mod
+	phis.oxy_mod /= weak_mod
+	phis.stamina_mod /= weak_mod
+	weak_mod = min(3, (1+(current_cycle*0.04)))
+	phis.brute_mod *= weak_mod
+	phis.burn_mod *= weak_mod
+	phis.oxy_mod *= weak_mod
+	phis.stamina_mod *= weak_mod
+	M.adjustToxLoss(-0.2*weak_mod) //Math is fun if you your PR doesn't accidentally get testmerged before you can test the effects of your equations!
 	..()
 	return TRUE
 
-/datum/reagent/medicine/C2/fiziver/on_mob_delete(mob/living/carbon/human/M)
+/datum/reagent/medicine/C2/fiziver/on_mob_delete(mob/living/carbon/human/M) //I was considering adding an on_mob_add counterpart to this, but it shouldn't ever be needed... right?
 	var/datum/physiology/phis = M.physiology
-	phis.brute_mod = max(phis.brute_mod - reset_mods, 1)
-	phis.burn_mod = max(phis.burn_mod - reset_mods, 1)
-	phis.oxy_mod = max(phis.oxy_mod - reset_mods, 1)
+	phis.brute_mod /= weak_mod //apparently, physiology stats are independent of species stats, so nothing bad should happen if someone changes race or something while this chem is in their system... hopefully
+	phis.burn_mod /= weak_mod
+	phis.oxy_mod /= weak_mod
```

Comment:
This is a lot worse than it used to be. weak_mod gets applied every tick but only gets removed once.

Reply:
Physiology was never meant to be used for fluent damage mods like this. The current system isn't capable of supporting what we are trying to do with it here.
Expand it or this will have to get fixed again later.

Reply:
Didn't see the removing part this is fine.

 The issue with this is that division causes a whole bunch of rounding errors. A proper solution might be one that uses signals to get the proper modifiers from multiple sources. Or something like that.